app/assets/stylesheets/application.scss

```diff
@@ -45,6 +46,11 @@
   }
 }
 
+.table .thead-light th {
+  color: $black;
+  font-weight: bold;
```

Comment:
Another way to achieve this would be to include Bootstrap utility classes in the table header